recipes/qbs/all/conanfile.py

```diff
@@ -0,0 +1,125 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class QbsConan(ConanFile):
+    name = "qbs"
+    description = "The Qbs build system"
+    topics = ("conan", "qbs", "build", "automation")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://qbs.io"
+    license = "LGPL-2.1-only", "LGPL-3.0-only", "Nokia-Qt-exception-1.1"
+    settings = "arch", "build_type", "compiler", "os"
+    no_copy_source=True
+    exports = ["patches/*.patch"]
+    build_policy = "missing"
+
+    build_requires = "qt/5.15.2"
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and self.settings.compiler == "Visual Studio":
+            self.build_requires("jom/1.1.3")
+
+    def configure(self):
+        if self.settings.os not in ["Windows", "Linux", "Macos"]:
+            raise ConanInvalidConfiguration("The OS ({}) is not supported by {}.".format(self.settings.os, self.name))
+        if self.settings.arch not in ["x86_64"]:
+            raise ConanInvalidConfiguration("The arch ({}) is not supported by {}.".format(self.settings.arch, self.name))
+
+        self.options["qt"].gui = False
+        self.options["qt"].openssl = False
+        self.options["qt"].qtscript = True
+        self.options["qt"].shared = False
+        self.options["qt"].widgets = False
+        self.options["qt"].with_odbc = False
+        self.options["qt"].with_pq = False
+        self.options["qt"].with_sqlite3 = False
+        self.options["qt"].with_zstd = False
```

Comment:
this cannot be done. if certain qt options absolutely need to have a given value, then what should be done instead is to check the value in `validate(self)` and raise `ConanInvalidConfiguration` if the value is incorrect

Reply:
@ericLemanissier, pardon, I don't understand, could you elaborate? This recipe requires a static Qt build with QtScript. Since it is static, I'd like to reduce Qt even further, but that is not strictly needed.

I tried this:
```
    def validate(self):
        if self.options["qt"].gui != False:
            raise ConanInvalidConfiguration("Option gui not supported")
        if self.options["qt"].qtscript != True:
            raise ConanInvalidConfiguration("Option qtscript is required")
        if self.options["qt"].widgets != False:
            raise ConanInvalidConfiguration("Option widgets not supported")
```
and got:
```
ERROR: qbs/1.18.0: Error in validate() method, line 54
	if self.options["qt"].gui != False:
	ConanException: option 'gui' doesn't exist
Possible options are []
```

Reply:
this could be because you are using qt as a build_requires whereas it is a requirement

Reply:
@ericLemanissier, now I am even more confused. Qt is only a build requirement for Qbs. Since this recipe does static linking, Qbs does not have any run-time dependencies on Qt.

Reply:
the fact that it links statically does not change the fact that qt is used at run-time by qbs. on the other hand, you probably want to use qbs as a build requirement.
Does qbs work if you remove all the checks on qt options ?

Reply:
> Does qbs work if you remove all the checks on qt options ?

It does not, because the prebuilt Qt packages on conan-center do not come with QtScript enabled. These are the minimum requirements I have:

        self.options["qt"].gui = False
        self.options["qt"].qtscript = True
        self.options["qt"].widgets = False

How would I implement that in the Qbs recipe?

> the fact that it links statically does not change the fact that qt is used at run-time by qbs.

Yes it is needed at run-time, but let me use different wording: There is no run-time dependency of Qbs on files from the Qt package. Qt is baked into the Qbs application binaries and thus, the Qbs package becomes 100% self-containing. Even if I linked Qt dynamically I would include the Qt libs into the Qbs package. The Qt package configuration used by Qbs must never ever leak to consumers of the Qbs package.

>  on the other hand, you probably want to use qbs as a build requirement.

Not probably. You always want to use Qbs as a build requirement. Would you ever want CMake as a (non-build) requirement in your project?

Reply:
unfortunately, CCI only builds the recipes with default options (the only variation being the `shared` option), so the only solution to get qbs in CCI would be to set the default value of qt:qtscript to True in the qt recipe first.
By the way, this dependency kind of makes QBS dead on arrival, because qtscript does not exist any more in qt6 :/

Reply:
> unfortunately, CCI only builds the recipes with default options (the only variation being the shared option), so the only solution to get qbs in CCI would be to set the default value of qt:qtscript to True in the qt recipe first.

that was the missing piece.

> By the way, this dependency kind of makes QBS dead on arrival, because qtscript does not exist any more in qt6 :/

You don't need to build Qbs with Qt6 in order to build Qt6 projects with it. In the meantime we are heavily working on a QJS port. Unfortunately the Qt company has abandoned QtScript without providing a sufficient replacement API, so porting is not trivial.

Reply:
> so the only solution to get qbs in CCI would be to set the default value of qt:qtscript to True in the qt recipe first.

@ericLemanissier, great, would that be acceptable? But maybe tomorrow somebody else needs qtdeclarative emabled. On the other hand, how would one ever bring a project to cci that needs Qt modules beyond qtbase?

Reply:
I have no idea if it's acceptable, or if CI time/disk space is sufficient to build qtbase+qtscript. You'd have to make a PR to know.